src/providers/vultr/mod.rs

```diff
@@ -0,0 +1,195 @@
+// Copyright 2020 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! vultr provider metadata fetcher
+//! This provider is selected via the platform ID `vultr`.
+//! The metadata endpoint is documented at https://www.vultr.com/metadata/.
+
+#[cfg(test)]
+use mockito;
+use openssh_keys::PublicKey;
+use slog_scope::{error, warn};
+use std::collections::HashMap;
+
+use crate::errors::*;
+use crate::network;
+use crate::providers::MetadataProvider;
+use crate::retry;
+
+#[cfg(test)]
+mod mock_tests;
+
+/// Provider prefix for Vultr Cloud.
+static PROVIDER_PREFIX: &str = "VULTR";
```

Comment:
I think this isn't really needed, you can directly hardcode the prefix in all places consuming this.